idarling/core/events.py

```diff
@@ -106,15 +106,15 @@ def __call__(self):
 class MakeDataEvent(Event):
     __event__ = "make_data"
 
-    def __init__(self, ea, flags, size, tid):
+    def __init__(self, ea, flags, size, sname):
         super(MakeDataEvent, self).__init__()
         self.ea = ea
         self.flags = flags
         self.size = size
-        self.tid = tid
+        self.sname = sname
 
     def __call__(self):
-        ida_bytes.create_data(self.ea, ida_bytes.calc_dflags(self.flags, True), self.size, self.tid)
+        ida_bytes.create_data(self.ea, ida_bytes.calc_dflags(self.flags, True), self.size, ida_struct.get_struc_id(self.sname) if self.sname else ida_idaapi.BADADDR)
```

Comment:
I agree it is better to forward the struct name than the struct id, but worth noting it will break old idarling servers that had the old way in the database.

Reply:
Hmm... Do you propose to implement the handling of old events? In the current case, this can be done easy. But, I am afraid, this will not always be the case.
Like variant, you can save IDB with downloaded changes locally and reupload it on server for getting database without olds events.

Reply:
No, but it is important for other people using it to know when we break backward compatibility. So here we will advise to do a new snapshot of all the databases before updating idarling. I think we could actually automate this, so will create a new issue